grails-app/controllers/org/pih/warehouse/product/CategoryController.groovy

```diff
@@ -57,7 +57,7 @@ class CategoryController {
         }
 
         if (!categoryInstance.hasErrors() && categoryInstance.save()) {
-            flash.message = "${warehouse.message(code: 'category.saved.message', arg: [format.category(category: categoryInstance)])}"
+            flash.message = "${warehouse.message(code: 'category.saved.message', arg: [format.category(category: categoryInstance).decodeHTML()])}"
```

Comment:
Here's another interesting case. We're taking the output of one taglib (`format.category`) and applying it to another (`warehouse.message`). In this case, the latter will html-encode the string it returns. If we don't call `decodeHTML()` on the argument to it, that part of the string will be double-escaped.

We don't need to call `decodeHTML()` on the outer taglib, because we're assigning it to flash.message, which must be html-escaped.